Sources/Build/BuildOperation.swift

```diff
@@ -334,16 +334,30 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
                 self.buildSystemDelegate?.preparationStepStarted(preparationStepName)
             }
             func didCompilePlugin(result: PluginCompilationResult) {
-                if !result.description.isEmpty {
-                    self.buildSystemDelegate?.preparationStepHadOutput(preparationStepName, output: result.description)
+                self.buildSystemDelegate?.preparationStepHadOutput(
+                    preparationStepName,
+                    output: result.commandLine.joined(separator: " "),
+                    verboseOnly: true
+                )
+                if !result.compilerOutput.isEmpty {
+                    self.buildSystemDelegate?.preparationStepHadOutput(
+                        preparationStepName,
+                        output: result.compilerOutput,
+                        verboseOnly: false
+                    )
                 }
                 self.buildSystemDelegate?.preparationStepFinished(preparationStepName, result: (result.succeeded ? .succeeded : .failed))
             }
             func skippedCompilingPlugin(cachedResult: PluginCompilationResult) {
                 // Historically we have emitted log info about cached plugins that are used. We should reconsider whether this is the right thing to do.
                 self.buildSystemDelegate?.preparationStepStarted(preparationStepName)
-                if !cachedResult.description.isEmpty {
-                    self.buildSystemDelegate?.preparationStepHadOutput(preparationStepName, output: cachedResult.description)
+                self.buildSystemDelegate?.preparationStepHadOutput(preparationStepName, output: cachedResult.compilerOutput, verboseOnly: false)
```

Comment:
Should this line be removed?  Otherwise this would emit the cached output twice?

Reply:
oops that is a copy past error